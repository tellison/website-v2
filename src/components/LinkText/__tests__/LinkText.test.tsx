import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import LinkText from '..';

describe('LinkText component', () => {

  it('internal link renders correctly', () => {
    const { container } = render(
      <LinkText
        href="/page1"
        children={'Page 1'}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('external URL renders correctly', () => {
    const { container } = render(
      <LinkText
        href="https://external-link.com"
        children={'External Link'}
      />
    );
    expect(container).toMatchSnapshot();
  });

});