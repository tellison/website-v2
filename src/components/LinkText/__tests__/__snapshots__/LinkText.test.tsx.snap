// Vitest Snapshot v1

exports[`LinkText component > external URL renders correctly 1`] = `
<div>
  <a
    href="https://external-link.com"
    rel="noreferrer"
    target="_blank"
  >
    External Link
  </a>
</div>
`;

exports[`LinkText component > internal link renders correctly 1`] = `
<div>
  <a
    href="/page1"
  >
    Page 1
  </a>
</div>
`;
